### DISCUSSION TITLE
Frigate records clips but webserver cannot playback H265 clips

### DISCUSSION THREAD

**Discussion by TheSleepySlee** [blakeblackshear/frigate#548] [Ask A Question] [answered]
Hi

Just switched to -rc1, however was facing issue before too.
Frigate works really great, and the new webserver UI is awesome.
However, I'm having issues because there seem to be a link problem between events and their videos.

Before installing rc1, I cleaned my complete media folders.
When I go the the webserver UI, all the events are displayed, and their associated thumbnails.
However impossible to play videos (whether they are H264 or H265).
If I look in my media folders manually, looking for the filename associated with the event ID, I can play the video fine using VLC.
I'm using NFS on local server to store frigate media.

What should I go to investigate this? Picture attached

<img width="464" alt="Capture d’écran 2021-01-19 à 18 17 39" src="https://user-images.githubusercontent.com/13090430/105070843-2782d280-5a84-11eb-8768-a0c437c3ba80.png">

**Comment by blakeblackshear** [comment, marked as answer]
What errors do you see in the browser logs?

**Reply by TheSleepySlee** [reply]
Failed to load resource: the server responded with a status of 403 (Forbidden)
<img width="557" alt="Capture d’écran 2021-01-19 à 18 33 59" src="https://user-images.githubusercontent.com/13090430/105071467-ee972d80-5a84-11eb-8af8-e350dda5dde3.png">

**Reply by blakeblackshear** [reply]
Sounds just like this issue: https://community.home-assistant.io/t/local-realtime-person-detection-for-rtsp-cameras/103107/3143?u=blakeblackshear

**Reply by TheSleepySlee** [reply]
Not sure... my permissions look ok, and NFS on local lan has R/W access.

```
root@docker:/opt/frigate# ll clips/beta/ |grep 1611071761.090809
-rwxrwxrwx 1 root root      20503 Jan 19 16:56 garage-1611071761.090809-cjfhqu.jpg*
-rwxrwxrwx 1 root root   16874051 Jan 19 16:56 garage-1611071761.090809-cjfhqu.mp4*
```

And why thumbnails are visible on this page: http://IP:5000/events ?
Where are they stored?

**Reply by paularmstrong** [reply]
> And why thumbnails are visible on this page: http://IP:5000/events ?

Thumbnails are base64-encoded data stored in the database and come back as a string that is embedded in the page; they don't come from the filesystem. You can see the data (kind of) in the bottom of your screenshot for `"thumbnail": "/9j/…`

**Reply by TheSleepySlee** [reply]
Issue still present on RC2.

**Reply by paularmstrong** [reply]
Sounds like the same issue that was posted here: https://github.com/blakeblackshear/frigate/issues/562#issuecomment-764215671

**Reply by TheSleepySlee** [reply]
Indeed. Thanks for pointing me there. But permissions look good on my server, even if the logs say the contrary.
Not sure what to try next.

```
2021/01/22 22:20:06 [error] 22#22: *6846 open() "/media/frigate/clips/z5-1611247066.847259-cn57ad.mp4" failed (13: Permission denied), client: 192.168.0.24, server: , request: "GET /clips/z5-1611247066.847259-cn57ad.mp4 HTTP/1.1", host: "192.168.0.121:5000", referrer: "http://192.168.0.121:5000/events/1611247066.847259-cn57ad"
2021/01/22 22:20:11 [warn] 22#22: *6846 an upstream response is buffered to a temporary file /var/lib/nginx/proxy/3/02/0000000023 while reading upstream, client: 192.168.0.24, server: , request: "GET /api/events?camera=z5&label=person HTTP/1.1", upstream: "http://127.0.0.1:5001/events?camera=z5&label=person", host: "192.168.0.121:5000", referrer: "http://192.168.0.121:5000/events?camera=z5&label=person"
2021/01/22 23:13:25 [warn] 21#21: *3 an upstream response is buffered to a temporary file /var/lib/nginx/proxy/1/00/0000000001 while reading upstream, client: 192.168.0.24, server: , request: "GET /api/garageptz/latest.jpg HTTP/1.1", upstream: "http://127.0.0.1:5001/garageptz/latest.jpg", host: "192.168.0.121:5000", referrer: "http://192.168.0.121:5000/"
2021/01/22 23:13:25 [warn] 21#21: *2 an upstream response is buffered to a temporary file /var/lib/nginx/proxy/2/00/0000000002 while reading upstream, client: 192.168.0.24, server: , request: "GET /api/z5/latest.jpg HTTP/1.1", upstream: "http://127.0.0.1:5001/z5/latest.jpg", host: "192.168.0.121:5000", referrer: "http://192.168.0.121:5000/"
2021/01/22 23:13:25 [warn] 21#21: *6 an upstream response is buffered to a temporary file /var/lib/nginx/proxy/3/00/0000000003 while reading upstream, client: 192.168.0.24, server: , request: "GET /api/japonais/latest.jpg HTTP/1.1", upstream: "http://127.0.0.1:5001/japonais/latest.jpg", host: "192.168.0.121:5000", referrer: "http://192.168.0.121:5000/"
2021/01/22 23:13:25 [warn] 21#21: *7 an upstream response is buffered to a temporary file /var/lib/nginx/proxy/4/00/0000000004 while reading upstream, client: 192.168.0.24, server: , request: "GET /api/portailcam/latest.jpg HTTP/1.1", upstream: "http://127.0.0.1:5001/portailcam/latest.jpg", host: "192.168.0.121:5000", referrer: "http://192.168.0.121:5000/"
```

```
root@51c6638cf126:/opt/frigate# ll /media/frigate/clips/z5-1611247066.847259-cn57ad.mp4
-rw-r--r-- 1 root root 7691232 Jan 21 17:38 /media/frigate/clips/z5-1611247066.847259-cn57ad.mp4
```

**Reply by blakeblackshear** [reply]
Check the permissions on the parent folder. They need to be 755.

**Reply by TheSleepySlee** [reply]
```
drwxr-xr-x   9 root root  4096 Jan 18 14:52 opt/
then
drwxr-xr-x  5 root root 4096 Dec 25 20:45 frigate/
then
drwxr-xr-x 4 1026 users 4096 Jan 19 15:03 clips/
then
drwxr-xr-x 3 root root  45056 Jan 22 18:44  beta/
then
-rw-r--r-- 1 root root 7691232 Jan 21 17:38 z5-1611247066.847259-cn57ad.mp4
```

I have tried to chown /clips to root, but did not solve.
Permissions are similar inside container.

**Reply by blakeblackshear** [reply]
Can you post your compose file again so I don't have to hunt through the other thread for it?

**Reply by TheSleepySlee** [reply]
Sure. Here it is below. What I find strange when you ask for permissions fix, is that frigate is able to write the clips. I have cleaned the remote /clips directory when I switched to -rc1.
If frigate is able to write clips, why is there permission issue when trying to read back?

Here is the fstab on the docker machine to mount nfs drive:
```
192.168.0.100:/volume2/camera/frigate /opt/frigate/clips nfs,user rw 0 0
```

and the docker compose:
```
version: "3.6"
services:
  frigate:
      container_name: frigate_beta
      restart: unless-stopped
      privileged: true
      image: blakeblackshear/frigate:0.8.0-rc3-amd64

      volumes:
        - /dev/bus/usb:/dev/bus/usb
        - /etc/localtime:/etc/localtime:ro
        - /opt/frigate/beta/config:/config
        - /opt/frigate/clips/beta:/media/frigate/clips
      ports:
        - "5000:5000"
        - "1935:1935" # RTMP feeds
      healthcheck:
        test: ["CMD", "wget" , "-q", "-O-", "http://localhost:5000"]
        interval: 30s
        timeout: 10s
        retries: 5
        start_period: 3m
```

**Reply by blakeblackshear** [reply]
Frigate (my python code) isn't reading the clips. Nginx is. The permissions issue is with nginx and it needs to have execute permissions on the folder where the clips are stored. Frigate doesn't need that permissions to write the files. It looks like it does have the right permissions, so I don't understand what is happening yet. I would see what `ls -lah /media/frigate/clips` says within the container. You can also try mounting the host directory to `/media/frigate` instead of `/media/frigate/clips` and let Frigate create the `clips` folder itself.

**Reply by TheSleepySlee** [reply]
```
root@f2bb45ca7f05:/opt/frigate# ls -lah /media/frigate/clips
total 7.0G
drwxr-xr-x 3 root root   44K Jan 23 13:44 .
drwxr-xr-x 4 root root  4.0K Jan 22 23:28 ..
drw-r--r-- 2 root users 4.0K Jan 19 18:16 @eaDir
-rw-r--r-- 1 root root   10M Jan 23 13:44 frigate.db
-rw-r--r-- 1 root root   16K Jan 19 15:55 garage-1611068103.682949-52dhkk.jpg
-rw-r--r-- 1 root root   19M Jan 19 15:55 garage-1611068103.682949-52dhkk.mp4
```
I will try what you suggest.

EDIT: recreated docker container, but no change.

New docker compose:
```
        - /opt/frigate/media:/media/frigate
```

New fstab:
```
192.168.0.100:/volume2/camera/frigate /opt/frigate/media nfs,user rw 0 0
```

New permissions:
```
root@8a140d92843d:/opt/frigate# ls -lah /media/frigate/clips
total 159M
drwxr-xr-x 2 root root 4.0K Jan 23 14:42 .
drwxr-xr-x 4 root root 4.0K Jan 23 14:32 ..
-rw-r--r-- 1 root root 440K Jan 23 14:42 frigate.db
-rw-r--r-- 1 root root  33K Jan 23 14:32 garageptz-1611408759.937055-due105.jpg
-rw-r--r-- 1 root root 8.7M Jan 23 14:33 garageptz-1611408759.937055-due105.mp4
```

New nginx logs:
```
root@8a140d92843d:/opt/frigate# cat /var/log/nginx/error.log |grep error 
root@8a140d92843d:/opt/frigate# cat /var/log/nginx/error.log            
2021/01/23 14:32:58 [warn] 20#20: *6 an upstream response is buffered to a temporary file /var/lib/nginx/proxy/1/00/0000000001 while reading upstream, client: 192.168.0.24, server: , request: "GET /api/garage/latest.jpg HTTP/1.1", upstream: "http://127.0.0.1:5001/garage/latest.jpg", host: "192.168.0.121:5000", referrer: "http://192.168.0.121:5000/"
2021/01/23 14:32:58 [warn] 20#20: *2 an upstream response is buffered to a temporary file /var/lib/nginx/proxy/2/00/0000000002 while reading upstream, client: 192.168.0.24, server: , request: "GET /api/z5/latest.jpg HTTP/1.1", upstream: "http://127.0.0.1:5001/z5/latest.jpg", host: "192.168.0.121:5000", referrer: "http://192.168.0.121:5000/"
2021/01/23 14:32:58 [warn] 20#20: *3 an upstream response is buffered to a temporary file /var/lib/nginx/proxy/3/00/0000000003 while reading upstream, client: 192.168.0.24, server: , request: "GET /api/portailcam/latest.jpg HTTP/1.1", upstream: "http://127.0.0.1:5001/portailcam/latest.jpg", host: "192.168.0.121:5000", referrer: "http://192.168.0.121:5000/"
```

**Reply by TheSleepySlee** [reply]
Here is swhat I have in JS console in chrome. Is that expected @paularmstrong ? thx
```
<video class="w-100" src="/clips/japonais-1611732985.254579-un3x9b.mp4" controls=""></video>
```

**Reply by TheSleepySlee** [reply]
@paularmstrong @blakeblackshear  I found the "bug". My cameras record H265, except one camera which is H264.
I can view the events & associated clips of the H264 camera just fine, but not all the other cameras.
Is that expected?
EDIT: changed title to reflect the actual issue.

**Reply by blakeblackshear** [reply]
h265 video is not well supported by browsers, so that is expected. Looks like they may play in Safari. https://caniuse.com/?search=hevc

**Reply by TheSleepySlee** [reply]
Tried in safari too, but with no luck. Need preview version.
